tools/developers/launch_instance.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "${AWS_USER}" ]; then
+    echo "Please set the AWS_USER environment variable"
+    exit 1
+fi
+
+AWS_SSH_KEYNAME="${AWS_USER}.ssh"
+AWS_REGION="us-east-2"
+: "${EC2_INSTANCE_SIZE:=t2.medium}"
+: "${SSH_PUBLIC_KEY_FILE:=~/.ssh/id_rsa.pub}"
+: "${SSH_PRIVATE_KEY_FILE:=~/.ssh/id_rsa}"
+# Default to Ubuntu Server 16.04 LTS (HVM), SSD Volume Type for us-east-2
+# https://aws.amazon.com/ec2/instance-types/
+: "${EC2_AMI:=ami-fcc19b99}"
+
+# expand the tilde
+eval SSH_PUBLIC_KEY_FILE=$SSH_PUBLIC_KEY_FILE
+eval SSH_PRIVATE_KEY_FILE=$SSH_PRIVATE_KEY_FILE
+
+BLUE='\033[0;34m'
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+YELLOW='\033[0;33m'
+CLEAR='\033[0m'
+
+echo_blue() {
+    echo -e "${BLUE}$1${CLEAR}"
+}
+
+echo_red() {
+    echo -e "${RED}$1${CLEAR}"
+}
+
+echo_green() {
+    echo -e "${GREEN}$1${CLEAR}"
+}
+
+echo_yellow() {
+    echo -e "${YELLOW}$1${CLEAR}"
+}
+
+app_banner() {
+cat << "EOF"
+   __  __        __  __ ___   _   ___
+ |  \/  |___ __|  \/  | __| /_\ / _ \
+ | |\/| / _ \_ / |\/| | _| / _ \ (_) |
+ |_|_ |_\___/__|_|  |_|___/_/ \_\___/
+ | __/ __|_  ) (_)_ _  __| |_ __ _ _ _  __ ___
+ | _| (__ / /  | | ' \(_-<  _/ _` | ' \/ _/ -_)
+ |___\___/___| |_|_||_/__/\__\__,_|_||_\__\___|
+ | |__ _ _  _ _ _  __| |_  ___ _ _
+ | / _` | || | ' \/ _| ' \/ -_) '_|
+ |_\__,_|\_,_|_||_\__|_||_\___|_|
+EOF
+echo ""
+}
+
+# ensure dependencies are on the PATH
+check_deps() {
+    set +e
+    echo "Checking for dependencies:"
+    echo -n "  aws... "
+    which aws > /dev/null
+    if [ $? -ne 0 ]; then
+        echo ""
+        echo_red "You don't seem to have the aws cli installed, or it's not on your path"
+        echo ""
+        echo "If you're on OSX, try:"
+        echo "brew install awscli"
+        echo "Please visit https://aws.amazon.com/cli/ for more info"
+        exit 1
+    else
+        echo_green '\o/'
+    fi
+
+    echo -n "  jq... "
+    which jq > /dev/null
+    if [ $? -ne 0 ]; then
+        echo ""
+        echo_red "You don't seem to have jq installed, or it's not on your path"
+        echo ""
+        echo "If you're on OSX, try:"
+        echo "brew install jq"
+        echo "Please visit https://stedolan.github.io/jq/ for more info"
+        exit 1
+    else
+        echo_green '\o/'
+    fi
+    set -e
+}
+
+# try to describe instances in a region
+# if it fails, ~/.aws/credentials is probably setup incorrectly
+aws_smoke_test() {
+    echo -n "Verifying AWS credentials work... "
+    aws ec2 describe-instances --region "${AWS_REGION}" > /dev/null 2>&1
+    if [ $? -eq 0 ]; then
+        echo_green '\o/'
+    else
+        echo ":-("
```

Comment:
And that's happening b/c `aws` command fails and script exits immediately due to `set -x`